kvdb/go.mod

```diff
@@ -23,6 +23,7 @@ require (
 	go.etcd.io/etcd/client/v3 v3.5.0
 	go.etcd.io/etcd/server/v3 v3.5.0
 	golang.org/x/net v0.0.0-20210405180319-a5a99cb37ef4
+	modernc.org/sqlite v1.20.0
```

Comment:
We'll likely need to drop some default platforms we build for w/ this. I don't think it'll be anything important tho

Reply:
update: managed to save the majority of these using the build flags. Let me know If we should save the rest too. 

With latest push, we now only drop `dragonfly-amd64`, `netbsd-386`, `netbsd-arm64` and `openbsd-386`
I can defs add the build tags for those too if needed.

Reply:
I think it's fine to drop these as they seem to be quite exotic. And dropping variants will speed up our release builds quite a bit.
If anyone complains about a missing combination, we can still add the build tags.

But I think we should document what operating systems or architectures are not supported to run SQLite, and why (in the `docs/sqlite.md` file).

And in the release notes we should mention that we are dropping support for these 4 os/arch combos.

Reply:
Another thing the `sqlite.md` doc should explain is that switching from bbolt to SQLite will result in a _new_ node being created. So migrating a node will only be possible once https://github.com/lightninglabs/lndinit/pull/21 is merged.

Reply:
> With latest push, we now only drop dragonfly-amd64, netbsd-386, netbsd-arm64 and openbsd-386
I can defs add the build tags for those too if needed.

Cool, I agree w/ oli above re these not being all that important in any case.

Reply:
cool - added the suggestions to the `sqlite.md` doc and to the release notes in the follow up PR